src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -65,7 +76,89 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // ToDo: Mingyi implment this method.
-        response.getWriter().println("You've called the post request");
+        String jsonBody = new BufferedReader(new InputStreamReader(request.getInputStream())).lines().collect(
+            Collectors.joining("\n"));
+
+        // Error handling: return error if jsonBody is empty
+        if (jsonBody == null || jsonBody.trim().length() == 0) {
+            System.out.println("the String fails to be parsed");
+            throw new NullPointerException();
+        }
+
+        // Parse the jsonBody String to a Gson object
+        Gson gson = new Gson();
+        Type type = new TypeToken<Map<String, String>>(){}.getType();
+        Map<String,String> eventJsonAsMap = gson.fromJson(jsonBody, type);
+
+        // Check if all the fields are present
+        if (!checkAllEventFieldsPresent(eventJsonAsMap)){
+            // throw error
+            System.out.println("Not all the fields in the map are present");
+            throw new NullPointerException();
```

Comment:
Can you throw an HTTP error instead and return the error message back to the frontend? Check out this https://stackoverflow.com/questions/839225/when-is-it-ok-to-throw-a-servletexception-from-a-servlet

Look into HTTP status code 400 Bad Request.